scanspec/service.py

```diff
@@ -104,10 +128,13 @@ def get_points(spec: Spec, max_frames: Optional[int] = 200000) -> PointsResponse
     """
     dims = spec.create_dimensions()  # Grab dimensions from spec
     path = Path(dims)  # Convert to a path
+
+    # TOTAL FRAMES
     total_frames = len(path)  # Capture the total length of the path
 
-    # Limit the consumed data to the max_frames argument
-    # # WARNING: path object is consumed after this statement
+    # MAX FRAMES | RETURNED FRAMES
+    # Limit the consumed data by the max_frames argument
+    # WARNING: path object is consumed after this statement
     if max_frames is None:
```

Comment:
No matter what the value of max_frames, we should be able to write:
```python
chunk = path.consume(max_frames)
returned_frames = len(chunk)
```
If we can't then path.consume is wrong...

Reply:
Tested in graphiql and seems to work as intended

Reply:
@MichaelStubbings I vaguely recall us talking about this `if max_frames is None` bit in the context of mypy complaining. Do you remember what the problem was?